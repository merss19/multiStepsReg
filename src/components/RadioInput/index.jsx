"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classNames = require("classnames");
var RadioInput = /** @class */ (function (_super) {
    __extends(RadioInput, _super);
    function RadioInput() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RadioInput.prototype.render = function () {
        var _a = this.props, input = _a.input, title = _a.title, name = _a.name, value = _a.value, _b = _a.meta, touched = _b.touched, error = _b.error;
        return (<span className={classNames('radio', { 'input--error': touched && !!error })}>
        <label className='radio__label' htmlFor={value}>
          <input {...input} className='radio__field' id={value} type='radio' name={name}/>
          <span className='radio__ph'/>
          <span>{title}</span>
        </label>
        {touched && error && <p className='input__alert'>{error}</p>}
			   </span>);
    };
    return RadioInput;
}(React.Component));
exports.default = RadioInput;
