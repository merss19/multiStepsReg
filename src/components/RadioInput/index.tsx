import * as React from 'react';
import { WrappedFieldProps } from 'redux-form';
import * as classNames from 'classnames';

interface OwnProps extends WrappedFieldProps<{}> {
  name: string
  title: string
  value: string
}

class RadioInput extends React.Component<OwnProps, {}> {
  render() {
    const {input, title, name, value, meta: {touched, error}} = this.props;
    return (
      <span className={classNames('radio', {'input--error': touched && !!error})}>
        <label
          className='radio__label'
          htmlFor={value}>
          <input
            {...input}
            className='radio__field'
            id={value}
            type='radio'
            name={name}
          />
          <span className='radio__ph'/>
          <span>{title}</span>
        </label>
        {touched && error && <p className='input__alert'>{error}</p>}
			   </span>
    );
  }
}

export default RadioInput;