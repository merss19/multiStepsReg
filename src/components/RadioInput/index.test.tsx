import {shallow, mount} from 'enzyme';
import RadioInput from './index';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let props = {
	name: 'gender',
	id: 'gender1',
	title: 'Мужчина',
		value: 'male',
	meta: {
		touched: true,
		error: 'Пол должен быть заполнен',
		asyncValidating: false,
		autofilled: false,
		invalid: false,
		pristine: false,
		dirty: false,
		dispatch: (fn: any) => fn,
		valid: false
	},
};
describe('Component RadioInput', () => {
	const component = shallow(<RadioInput {...props}/>);
	it('renders component', () => {
		expect(component).toHaveLength(1);
	});
	it('renders correctly', () => {
		const tree = toJson(shallow(<RadioInput {...props}/>));
		expect(tree).toMatchSnapshot();
	})
});

describe('Component RadioInput DOM', () => {
	const component = mount(<RadioInput {...props}/>);
	describe('render props', () => {
		it('render name', () => {
			expect(component.props().name).toBe(props.name);
		});
		it('render id', () => {
			expect(component.props().id).toBe(props.id);
		});
		it('render title', () => {
			expect(component.props().title).toBe(props.title);
		});
		it('do not calls setRadioError, touched = false', () => {
			let props2 = {
				name: 'gender',
				id: 'gender1',
				title: 'Мужчина',
				setRadioError: jest.fn(),
				removeRadioError: jest.fn(),
				meta: {
					touched: false,
					error: 'Пол должен быть заполнен',
					asyncValidating: false,
					autofilled: false,
					invalid: false,
					pristine: false,
					dirty: false,
					dispatch: (fn: any) => fn,
					valid: false
				},
			};
			component.setProps({props2});
			expect(props2.setRadioError).not.toHaveBeenCalled();
		});
		it('calls removeRadioError, error = ""', () => {
			let props3 = {
				name: 'gender',
				id: 'gender1',
				title: 'Мужчина',
				setRadioError: jest.fn(),
				removeRadioError: jest.fn(),
				meta: {
					touched: true,
					error: '',
					asyncValidating: false,
					autofilled: false,
					invalid: false,
					pristine: false,
					dirty: false,
					dispatch: (fn: any) => fn,
					valid: false
				},
			};
			component.setProps({ props3});
			expect(props3.removeRadioError).not.toHaveBeenCalled();
			expect(props3.setRadioError).not.toHaveBeenCalled();
		})
	})
});
