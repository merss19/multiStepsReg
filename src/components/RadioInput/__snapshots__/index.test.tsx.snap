// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component RadioInput renders correctly 1`] = `
<span
  className="radio input--error"
>
  <label
    className="radio__label"
    htmlFor="male"
  >
    <input
      className="radio__field"
      id="male"
      name="gender"
      type="radio"
    />
    <span
      className="radio__ph"
    />
    <span>
      Мужчина
    </span>
  </label>
  <p
    className="input__alert"
  >
    Пол должен быть заполнен
  </p>
</span>
`;
